interaction draft (#353)
malcolmbarrett/causal-inference-in-R: chapters/14-interaction.qmd

Consider a simple example: Does a new medication reduce blood pressure, and does this effect differ between men and women?
We might fit a model like `blood_pressure ~ medication + sex + medication:sex`.
The coefficient on the interaction term tells us something, but what exactly?
Does it tell us about how the medication's effect varies by sex?
Or does it tell us about how the joint effect of medication and sex differs from their separate effects?
These are different questions, and the answer depends on whether we're thinking about **effect modification** or **interaction**.

This distinction matters because effect modification and interaction are fundamentally different concepts, even though they're often used interchangeably [@VanderWeele2009].
Understanding when they coincide and when they don't is crucial for correctly interpreting our models and drawing valid causal conclusions.
In this chapter, we'll use causal diagrams to clarify these concepts and show how to assess each appropriately.

## Effect modification versus interaction

> **Effect modification** occurs when the effect of one exposure (E) on an outcome (D) varies across strata of a third variable (Q).

Reviewer: ```suggestion
**Effect modification** occurs when the effect of one exposure (X) on an outcome (Y) varies across strata of a third variable (Q).
```

For example, there may be effect modification on the additive scale (risk difference) but not on the multiplicative scale (risk ratio), or vice versa.
When the effects are in opposite directions across strata of Q, we have **qualitative effect modification**.
In this case, effect modification will be present on both additive and multiplicative scales [@hernan2021].

It's important to note that effect modification by Q does not necessarily imply that Q plays a causal role in modifying the effect.
Q may be a **surrogate effect modifier**, that is, a marker for the true causal effect modifier.
For example, nationality might modify the effect of treatment because it's correlated with quality of care, which is the true causal effect modifier [@hernan2021].

**Interaction** on the causal risk difference scale occurs when:
$$E[D_{e_1 q_1}] - E[D_{e_0 q_1}] \neq E[D_{e_1 q_0}] - E[D_{e_0 q_0}]$$

Here, we're intervening on both E and Q (looking at $D_{eq}$).
This definition asks: does the effect of E vary depending on what level we set Q to?

Because interaction is symmetric, this definition is equivalent to asking whether the effect of Q varies depending on what level we set E to:

> $$E[D_{e_1 q_1}] - E[D_{e_1 q_0}] \neq E[D_{e_0 q_1}] - E[D_{e_0 q_0}]$$

Reviewer: ```suggestion
$$E[Y(1, q_1)] - E[Y(1, q_0)] \neq E[Y(0, q_1)] - E[Y(0, q_0)]$$
```

These definitions make clear that effect modification and interaction are asking different questions.
Effect modification asks about variation in the effect of E across observed levels of Q.
Interaction asks about variation in the effect of E across intervened-upon levels of Q (or equivalently, variation in the effect of Q across intervened-upon levels of E).

## Representing interaction and effect modification in DAGs

As noted earlier, conventional DAGs don't inherently represent interaction or effect modification, though some proposals exist to include these features.
Here we demonstrate three approaches using a simple example where E (treatment) and Q (another exposure) may interact in their effect on D (outcome), with X as a confounder.

### Conventional DAGs

The standard approach is to draw a conventional DAG without explicitly representing interactions.

```{r}
#| label: fig-dag-conventional
#| fig-cap: "A conventional DAG showing E, Q, and D without explicitly representing interaction. The interaction, if present, is implicit in the functional form of the relationship."
#| fig-width: 5
#| fig-height: 4

conventional_dag <- dagify(
  E ~ X,
  Q ~ X,

>   D ~ E + Q + X,

Reviewer: ```suggestion
  Y ~ X + Q + Z,
```

  coords = time_ordered_coords(
    list(
      c("X"),
      c("E", "Q"),
      "D"
    )
  ),
  exposure = "E",

>   outcome = "D",

Reviewer: ```suggestion
  outcome = "Y",
```

For most purposes, conventional DAGs suffice, with interactions handled in the statistical modeling stage.

## When effect modification and interaction differ

It's possible to have effect modification without interaction, and vice versa.
Understanding when and why this happens helps clarify the distinction.

### Effect modification without interaction

Consider the DAG in @fig-effect-mod-no-interaction, which uses `ggdag` to reproduce an example given by @VanderWeele2009.
Here, E is a drug, D is hypertension, X is a person's genotype, and Q is hair color.
Hair color serves as a proxy for a genotype that drives risk of hypertension.

```{r}
#| label: fig-effect-mod-no-interaction
#| fig-cap: "A DAG showing effect modification by Q (hair color) of the effect of E (drug) on D (hypertension) without interaction between E and Q. Hair color Q is a proxy for genotype X, which modifies the effect of E on D."
#| fig-width: 5
#| fig-height: 4

effect_mod_dag <- dagify(
  D ~ E + X,
  Q ~ X,
  coords = time_ordered_coords(
    list(
      c("X"),
      c("E", "Q"),
      "D"
    )
  ),
  exposure = "E",
  outcome = "D",
  labels = c(
    E = "Drug",
    D = "Hypertension",

>     X = "Genotype",

Reviewer: ```suggestion
    Z = "Genotype",
```

This example illustrates that a variable can be an effect modifier without there being an interaction between the effects of that variable and the primary exposure.

### Interaction without effect modification

The converse is also possible, where interaction can exist without effect modification.
Consider @fig-interaction-no-effect-mod which is another `ggdag` reporoduction from [@VanderWeele2009].
E is a weight-loss drug given in a randomized trial, D is weight at the end of the trial, X is sugar intake, and Q is exercise level.

```{r}
#| label: fig-interaction-no-effect-mod
#| fig-cap: "A DAG showing potential interaction between the effects of E (weight-loss drug) and Q (exercise) on D (weight) without effect modification by Q of the effect of E on D. Sugar intake X affects both exercise Q and weight D."
#| fig-width: 5
#| fig-height: 4

interaction_dag <- dagify(
  Q ~ X,
  D ~ X + Q + E,
  coords = time_ordered_coords(
    list(
      c("X"),
      c("E", "Q"),
      "D"
    )
  ),
  exposure = "E",
  outcome = "D",
  labels = c(
    E = "Weight-loss\ndrug",
    D = "Weight",

>     X = "Sugar\nintake",

Reviewer: ```suggestion
    Z = "Sugar\nintake",
```

It's therefore possible that we observe $E[D_{e_1} | Q = q_1] - E[D_{e_0} | Q = q_1] = E[D_{e_1} | Q = q_0] - E[D_{e_0} | Q = q_0]$, meaning Q is not an effect modifier, even though there is interaction between the effects of E and Q when we intervene on both.

The key insight is that conditioning on Q (as in effect modification) captures information about common causes of Q and D (like X), while intervening on Q (as in interaction) does not.
This difference can lead to effect modification and interaction giving different answers.

Let's demonstrate this with a simulation:

```{r}
#| message: false
#| warning: false
set.seed(67)
library(tidyverse)
library(broom)

# Simulate data where:
# - X (sugar intake) affects both Q (exercise) and D (weight)
# - E (drug) and Q interact in their effect on D
# - But conditioning on Q masks the interaction due to confounding by X
# We set up the parameters so that when we condition on Q without controlling for X,
# the confounding cancels out the interaction, making it appear there's no effect modification

n <- 2000
sim_data <- tibble(
  # Sugar intake (confounder)
  X = rnorm(n),
  # Exercise level (affected by sugar intake)

>   Q = rbinom(n, 1, plogis(-0.5 + 1.2 * X)),

Reviewer: ```suggestion
  Q = rbinom(n, 1, plogis(-0.5 + 1.2 * Z)),
```

```{r}
#| message: false
# Simulate data where exercise and diet interact
# Use plogis to ensure probabilities stay in [0,1]
# We set up strong confounding for diet to make the difference between
# effect modification and interaction pronounced
set.seed(123)
n <- 1000
sim_data2 <- tibble(
  confounder = rnorm(n),
  exercise = rbinom(n, 1, plogis(-1 + 0.5 * confounder)),
  diet = rbinom(n, 1, plogis(-0.5 + 2 * confounder)),  # Very strong confounding for diet
  weight_loss = 5 * exercise + 4 * diet + 3 * exercise * diet + 5 * confounder + rnorm(n, 0, 3)
)

# If both effects are unconfounded given confounder, we can assess interaction
mod_interaction <- lm(weight_loss ~ exercise + diet + exercise:diet + confounder, data = sim_data2)
tidy(mod_interaction) |>
  filter(str_detect(term, "exercise|diet"))
```

The interaction term `exercise:diet` is `r round(coef(mod_interaction)[4], 2)`, suggesting that the joint effect of exercise and diet together is greater than the sum of their individual effects.
If we've correctly controlled for confounding of both exercise and diet, this coefficient measures interaction.

### Marginal structural models

> **Marginal structural models (MSMs)** provide a framework for distinguishing between effect modification and interaction in the analysis [@VanderWeele2009].

Reviewer 2: I feel like we need a callout box for the term marginal structural models which is used to refer to many techniques, presumably because it is a generalization of them (but I haven't thought it through or looked into usage all that much)

They do this through different weighting schemes.

For effect modification, we use inverse probability of treatment weighting (IPTW) for E only.
The weights are:
$$w^E = \frac{P(E = e | Q = q)}{P(E = e | Q = q, X = x)}$$

We then fit a weighted regression model:
$$E[D_e | Q = q] = \alpha_0 + \alpha_1 e + \alpha_2 q + \alpha_3 e \times q$$

The parameter $\alpha_3$ measures effect modification.

For interaction, we need weights for both E and Q:
$$w^E = \frac{P(E = e | Q = q)}{P(E = e | Q = q, X = x)}$$

> $$w^Q = \frac{P(Q = q)}{P(Q = q | X = x)}$$

Reviewer 2: I think we need some intuition here, given the symmetric nature of interaction you lay out, about why we have 

$$w^Q = \frac{P(Q = q)}{P(Q = q | X = x)}$$

and not 

$$w^Q = \frac{P(Q = q)}{P(Q = q | E = e, X = x)}$$
